includes/class-migrator-cli-payment-methods.php

```diff
@@ -68,12 +78,230 @@ private function search_payment_token_by_stripe_id( $saved_payment_tokens, $stri
+		$tokens = WC_Payment_Tokens::get_customer_tokens( $user_id );
+
+		foreach ( $tokens as $token ) {
+			if ( $token->get_meta( self::ORIGINAL_PAYMENT_METHOD_ID_KEY ) === $old_payment_method_id ) {
+				if ( (int) $old_payment_method_last_4 === (int) $token->get_meta( 'last4' ) ) {
+					return $token;
+				} else {
+					WP_CLI::line( WP_CLI::colorize( '%RMissmatch Payment Token last 4:%n' ) . $old_payment_method_id );
+				}
+			}
+		}
+	}
+
+	/**
+	 * Searches a WP_User by the stripe id.
+	 *
+	 * @param $stripe_id
+	 * @return WP_User
+	 * @throws Exception
+	 */
+	private function get_user_by_stripe_id( $stripe_id ) {
+		$users = get_users(
+			array(
+				'meta_key'   => 'wp_' . self::get_customer_id_option(),
+				'meta_value' => $stripe_id,
+			)
+		);
+
+		if ( is_wp_error( $users ) || ! $users ) {
+			WP_CLI::line( WP_CLI::colorize( '%RCustomer not found%n' ) );
+			return;
+		}
+
+		if ( count( $users ) > 1 ) {
+			WP_CLI::line( WP_CLI::colorize( '%RMultiple Customers found%n' ) );
+			return;
+		}
+
+		return reset( $users );
+	}
+
+	/**
+	 * Sets the new payment method for orders and subscriptions
+	 *
+	 * @param string $type shop_order|shop_subscription.
+	 */
+	private function update_orders_or_subscriptions( $type ) {
+		$page = 1;
+
+		do {
+			$orders = wc_get_orders(
+				array(
+					'type'   => $type,
+					'status' => 'all',
+					'limit'  => 100,
+					'paged'  => $page,
+				)
+			);
+
+			/** @var WC_Order|WC_Subscription $order */
+			foreach ( $orders as $order ) {
+				WP_CLI::line( 'Processing ' . $type . ': ' . $order->get_id() );
+
+				switch ( $order->get_meta( self::ORIGINAL_PAYMENT_GATEWAY_KEY ) ) {
+					case 'shopify_payments':
+						$this->process_shopify_payments( $order );
+						break;
+					case null:
+						WP_CLI::line( WP_CLI::colorize( ' %RPayment gateway not set%n ' ) );
+						break;
+					default:
+						WP_CLI::line( WP_CLI::colorize( '%RUnkown payment gateway:%n ' ) . $order->get_meta( self::ORIGINAL_PAYMENT_GATEWAY_KEY ) );
+				}
+			}
+
+			++$page;
+		} while ( $orders );
+	}
+
+	/**
+	 * Returns the meta_key where the stripe customer_id is stored without the wp_ at the beginning.
 	 * Extracted from https://github.com/Automattic/woocommerce-payments/blob/92525c2a637bf592ec412bb0a979ab91862575d1/includes/class-wc-payments-customer-service.php#L407-L416
+	 * @return string
+	 * @throws Exception
 	 */
-	private function get_customer_id_option(): string {
+	private static function get_customer_id_option(): string {
 		return WC_Payments::mode()->is_test()
 			? WC_Payments_Customer_Service::WCPAY_TEST_CUSTOMER_ID_OPTION
 			: WC_Payments_Customer_Service::WCPAY_LIVE_CUSTOMER_ID_OPTION;
 	}
+
+	/**
+	 * Updates shopify_payments to WooPayments.
+	 *
+	 * @param WC_Order|WC_Subscription $order the order or subscription to be updated
+	 * @throws WC_Data_Exception
+	 */
+	private function process_shopify_payments( WC_Order|WC_Subscription $order ) {
```

Comment:
I got a critical error on my test site running PHP 7.4. I looked up and found that the union type declaration within the function param would work in PHP 8.0+ . It may cause backward compatibility issue with sites running versions less than PHP 8.0.

We could add an `if` statement to check for Order / Subscription. 

```suggestion
	private function process_shopify_payments( $order ) {
	    if ( ! $order instanceof WC_Order || ! $order instanceof WC_Subscription ) {
			 WP_CLI::line( WP_CLI::colorize( '%RInvalid order type%n' ) );
			 return;
		 }
          .....
          ....
         }
```

Reply:
I only added that for IDE hints. So I'll just remove it. A subscription is a WC_Order too so it should be ok.

Reply:
That makes sense. Thanks for addressing this 👍